Custom timeout does not take effect when provisioning a EHA cluster

### Terraform Version
Terraform v1.3.4

### Affected Resource(s)
- cluster_resource

### Debug Output

```
2022-11-18T18:17:09.761+0800 [INFO]  provider.terraform-provider-biganimal: 2022/11/18 18:17:09 [TRACE] Waiting 10s before next try: timestamp=2022-11-18T18:17:09.761+0800
biganimal_cluster.this_resource: Still creating... [19m0s elapsed]
2022-11-18T18:17:19.618+0800 [INFO]  provider.terraform-provider-biganimal: 2022/11/18 18:17:19 [WARN] WaitForState timeout after 19m0s: timestamp=2022-11-18T18:17:19.618+0800
2022-11-18T18:17:19.618+0800 [INFO]  provider.terraform-provider-biganimal: 2022/11/18 18:17:19 [WARN] WaitForState starting 30s refresh grace period: timestamp=2022-11-18T18:17:19.618+0800
2022-11-18T18:17:19.620+0800 [ERROR] provider.terraform-provider-biganimal: Response contains error diagnostic: @module=sdk.proto tf_proto_version=5.3 tf_req_id=e41f0d95-d731-ab1f-bf5d-21ffa6ca4b69 tf_rpc=ApplyResourceChange @caller=/Users/kunliu/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/diag/diagnostics.go:55 diagnostic_detail= diagnostic_severity=ERROR diagnostic_summary="Instance not yet ready" tf_provider_addr=registry.terraform.io/hashicorp/biganimal tf_resource_type=biganimal_cluster timestamp=2022-11-18T18:17:19.619+0800
╷
│ Error: Instance not yet ready
│ 
│   with biganimal_cluster.this_resource,
│   on resource.tf line 20, in resource "biganimal_cluster" "this_resource":
│   20: resource "biganimal_cluster" "this_resource" {
│ 
╵
2022-11-18T18:17:19.648+0800 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2022-11-18T18:17:19.651+0800 [DEBUG] provider: plugin process exited: path=/Users/kunliu/.terraform.d/plugins/hashicorp.com/edu/biganimal/0.3.1/darwin_arm64/terraform-provider-biganimal pid=83377
2022-11-18T18:17:19.651+0800 [DEBUG] provider: plugin exited

```

### Expected Behavior
The EHA cluster should be provisioned successfully

### Actual Behavior
The issue was initial observed by @pizzacapricciosa, when provisioning a EHA cluster, it sometimes need more than 20mins, terraform got time out error after 20mins. I looked through the code and found the custom timeout has been set to [45mins](https://github.com/EnterpriseDB/terraform-provider-biganimal/blob/467cf215f5b8c1436573ad48cc9fc049704fa6d9/pkg/provider/resource_cluster.go#L32). But somehow it doesn't take effect. 

### Steps to Reproduce
Provisioning a EHA cluster

--- a/pkg/provider/resource_cluster.go
+++ b/pkg/provider/resource_cluster.go
@@ -256,7 +256,7 @@ func (c *ClusterResource) Create(ctx context.Context, d *schema.ResourceData, me
 	// retry until we get success
 	err = resource.RetryContext(
 		ctx,
-		d.Timeout(schema.TimeoutUpdate)-time.Minute,
+		d.Timeout(schema.TimeoutCreate)-time.Minute,
 		c.retryFunc(ctx, d, meta, clusterId))
 	if err != nil {
 		return diag.FromErr(err)

--- a/pkg/provider/resource_region.go
+++ b/pkg/provider/resource_region.go
@@ -35,7 +35,7 @@ func (r *RegionResource) Schema() *schema.Resource {
 		Timeouts: &schema.ResourceTimeout{
 			Create: schema.DefaultTimeout(60 * time.Minute),
 			Update: schema.DefaultTimeout(60 * time.Minute),
-			Delete: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
 		},
 
 		Schema: map[string]*schema.Schema{

